### PR TITLE
Improved error display

### PR DESCRIPTION
**Pull request by juxd**
(Ignore the branch name, it's misleading)

Should be sufficient to handle:
- Join room when not in `FINISHED` state
- No more pins
- Multiple connection per device
- Timeouts

@hzxcaryn my error display component looks garbo now, sorry :(

### Known kinks to iron out
- [x] Navigating back and forth the form probably shouldn't let the error linger
- [x] Permission errors are not integrated - perhaps they should at least have thee same styling?

### CODE REVIEW

**Comment by juxd** [comment]
I realise the `JoinRoomPage` and `CreateRoomPage` components aren't very DRY - can get to refactoring that soon too actually.

File: Web/src/components/join-room/JoinRoomPage.tsx [hunk 1]
```diff
@@ -21,9 +23,9 @@ const JoinRoomContainer = styled.div`
   text-align: center;
 
   h1 {
-    font-family: var(--primary);
-    font-size: 1.4rem;
-    font-weight: normal;
+  font-family: var(--primary);
```
Comment (indocomsoft) [inline Web/src/components/join-room/JoinRoomPage.tsx]: Looks like wrong indentation heh

File: Web/src/components/room/ErrorDisplay.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  error: CommError | null;
+  errorDescription: string | null;
+}
+
+interface TitleMessage {
+  title: string;
+  message?: string;
+}
+
+const titleMessageMapping: { [error in CommError]: TitleMessage } = {
```
Comment (indocomsoft) [inline Web/src/components/room/ErrorDisplay.tsx]:
```suggestion
const titleMessageMapping: Record<CommError, TitleMessage> = {
```

Nit: more descriptive type lol

File: Web/src/components/create-room/CreateRoomPage.tsx [hunk 3]
```diff
@@ -95,6 +104,20 @@ const CreateRoomPage: React.FC<{ commHooks: CommAttributes }> = ({ commHooks })
             Unable to get permissions to play Pinda.
           </ErrorText>
         )}
+      {waitingForResponse
+        && (
+          <ReactLoading
```
Comment (indocomsoft) [inline Web/src/components/create-room/CreateRoomPage.tsx]: Not sure why not use our common `Loading` component?

File: Web/src/components/room/ErrorDisplay.tsx (hunk 2 above)
Comment (indocomsoft) [reply Web/src/components/room/ErrorDisplay.tsx]: https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkt

It’s the same as what you had previously, but I think this looks clearer

File: Web/src/components/join-room/JoinRoomPage.tsx (hunk 1 above)
Comment (hzxcaryn) [inline Web/src/components/join-room/JoinRoomPage.tsx]:
```suggestion
    font-family: var(--primary-font);
```

File: Web/src/components/create-room/CreateRoomPage.tsx (hunk 3 above)
Comment (juxd) [reply Web/src/components/create-room/CreateRoomPage.tsx]: Hmm, I just wanted to show like progress on the same page to indicate that there's some operations running currently. Should we change it to the loading component, or remove completely?

ping @hzxcaryn

Comment (indocomsoft) [reply Web/src/components/create-room/CreateRoomPage.tsx]: I think it's fine to do this then, since I assume it's not really the full screen?

**Comment by juxd** [comment]
Yeah, i'm planning to do that!

**Pull request merged**